src/fhirtypes/common.ts

```diff
@@ -218,6 +218,28 @@ export function setPropertyOnInstance(
         } else {
           if (typeof assignedValue === 'object') {
             Object.assign(current[key][index], assignedValue);
+            // check if we have used modifier extensions correctly
+            if ((key === 'extension' || key === 'modifierExtension') && assignedValue.url != null) {
+              const extension = fisher.fishForFHIR(assignedValue.url, Type.Extension);
+              if (extension) {
+                const isModifierExtension =
+                  extension.snapshot.element.find((el: ElementDefinition) => el.id === 'Extension')
+                    .isModifier === true;
```

Comment:
I _think_ this same check is used in all three places, so maybe this could be a separate function?